ui/modal/password_modal.go

```diff
@@ -168,6 +171,7 @@ func (pm *PasswordModal) Handle() {
 	if pm.modal.BackdropClicked(pm.isCancelable) {
 		if !pm.isLoading {
 			pm.Dismiss()
+			pm.onBackdropClicked()
 		}
 	}
 }
```

Comment:
This already implements the callback needed.. Clicking the backdrop is similar to pressing the cancel button.
```suggestion
	if pm.modal.BackdropClicked(pm.isCancelable) {
		if !pm.isLoading {
			pm.Dismiss()
			pm.negativeButtonClicked()
		}
	}
}
```